modules/ROOT/pages/debugging-outside-studio.adoc

```diff
@@ -0,0 +1,40 @@
+= Debugging Outside Studio
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: debugging, debug
+
+
+To troubleshoot a Mule app in Anypoint Studio, you can use the xref:studio::visual-debugger-concept.adoc[Studio Visual Debugger]. This lets you set breakpoints to stop execution and check events.
+
+If you're creating your Mule app outside of Studio, you can either use MUnit to test your app, or you can activate remote debugging.
+
+== Use MUnit to Test an App
+
+MUnit is a Mule application testing framework that allows you to build automated tests for your integrations and APIs. It provides a full suite of integration and unit test capabilities, and is fully integrated with Maven and Surefire for integration with your continuous deployment environment.
+
+For more information, see the MUnit documentation.
+
+== Activate Remote Debugging
```

Comment:
Just a question here, we want people to debug java classes of the mule runtime only? Do we want to also provide a guide to remote debug the mule application using the Studio canvas as well?

Reply:
I have made Fer aware of this so he can add it to the Studio doc if it belongs there. We can link to it from this topic once the topic is available. (I'll create a JIRA for this.)

Can you approve the topic now? Thanks for the review.